### PR TITLE
Updated Travis scripts

### PR DESCRIPTION
**Pull request by clintonb**
- We now pull images prior to building. This helps avoid wasting time completely rebuilding the app container.
- Added test to ensure migrations have been generated, and run properly.
- We wait for the database to become available before running Django management commands

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,6 +29,9 @@ clean: ## Delete generated byte code and coverage reports
 	find . -name '*.pyc' -delete
 	coverage erase
 
+detect_missing_migrations:  ## Determine if any apps are missing generated migrations
```
Comment (r-b-g-b) [inline Makefile]: This exists already, no? https://github.com/openoakland/woeip/blob/82e36bf5c4b0806c5fd32dd3a5dcc0844aee9d3d/Makefile#L55

File: .travis.yml [hunk 2]
```diff
@@ -10,11 +10,15 @@ cache: pip
 
 install:
   - bash .travis/install_terraform.sh
+  - make docker.pull
   - make docker.build
 
 script:
+  - docker-compose -f docker-compose.yml -f docker-compose.travis.yml up -d db
+  - docker-compose -f docker-compose.yml -f docker-compose.travis.yml run app "make detect_missing_migrations"
```
Comment (r-b-g-b) [inline .travis.yml]: Is there any hard check that the postgis service is initialized and ready to accept connections, or does this just hope that doing ``up -d db`` first will give it enough time to start before the ``detect_missing_migrations``.

https://travis-ci.org/openoakland/woeip/builds/469069161#L1022

Seems to need a hard check solution:
- [wait-for-it](https://github.com/vishnubob/wait-for-it/blob/master/wait-for-it.sh)
- [good discussion here](https://github.com/docker-library/postgres/issues/146)

**Comment by TangoYankee** [comment]
What do you think is causing: ```psycopg2.OperationalError: could not connect to server: Connection refused```?

**Comment by clintonb** [comment]
The database container hasn’t started. I thought I’d resolved this.

**Comment by TangoYankee** [comment]
Yeah, my understanding is still too limited to make a meaningful contribution

**Comment by clintonb** [comment]
This is ready. @TangoYankee or @r-b-g-b, please approve so that I can merge.

**Comment by r-b-g-b** [comment]
Nice! Looks like it's behaving better now. One issue I read about involves wait-for-it.sh and the way the postgres docker containers are instantiated. wait-for-it.sh just checks that a given host:port responds. While initializing itself, the postgres container starts and stops several times before it is actually ready. It is possible that wait-for-it.sh falsely detects the postgres service during one its intermediate phases. No sense for how likely this is to ever occur, just something to keep in mind if we ever notice random failed tests. I think this looks ready to merge, thanks @clintonb !

**Review by r-b-g-b — APPROVED** [review]

**Comment by clintonb** [comment]
@r-b-g-b your concern is valid in general, but not an issue in this specific case. The restart issues only apply on the first run of the database. The first line of `script` starts both containers. We don't actually need database access until the calls to the `migrate` and `make` targets. The Postgres container initialization is relatively quick, and is completed by the time we reach those lines in the script.

I imagine this race condition will be fine for the foreseeable future. If any drastic changes are made to that container, we should definitely revisit this issue.

**Pull request merged**